libr/core/canal.c

```diff
@@ -2525,6 +2525,33 @@ static int fcn_print_legacy(RCore *core, RAnalFunction *fcn) {
 	RList *refs, *xrefs;
 	int ebbs = 0;
 	char *name = r_core_anal_fcn_name (core, fcn);
+
+	// execution times:
+	r_cons_printf ("execution state: ");
+	if (core->dbg->trace->enabled == 0) {
+		r_cons_printf ("(dbg.trace unset)\n");
+	} else if (core->dbg->trace->enabled == 1) {
+		unsigned long long max_addr = fcn->addr;
+		max_addr += (ut64)r_anal_fcn_size (fcn);
+
+		bool flag = false;
+		int tag = core->dbg->trace->tag;
+		RListIter *iter;
+		RDebugTracepoint *trace;
+		r_list_foreach (core->dbg->trace->traces, iter, trace) {
+		  if (!trace->tag || (tag & trace->tag)) {
+			if (trace->addr >= fcn->addr && trace->addr <= max_addr) {
```

Comment:
thats not how you c heck if an offset is inside a function. there's an api  for that. functions are not  linear